packages/builder/src/steps/invoke.ts

```diff
@@ -95,6 +95,13 @@ async function runTxn(
     overrides.maxPriorityFeePerGas = runtime.priorityGasFee;
   }
 
+  // Attempt to encode data so that if any arguments have any type mismatches, we can catch them and present them to the user.
+  try {
+    contract.interface.encodeFunctionData(config.func, config.args);
+  } catch (error: any) {
+    throw new Error(`Invalid arguments for function "${config.func}": \n\n ${error}`);
```

Comment:
we may be able to inspect the data structure of `error` to see if it includes any additional useful data we can pretty print